File: since2020/CodeForces/1328B.cpp
/**
 *  @judge CodeForces
 *  @id 1328B
 *  @name K-th Beautiful String
 *  @contest Codeforces Round #629 (Div. 3)
 * 
 *  @tag Combination & Permutaion
 */
#include <bits/stdc++.h>
using namespace std;

typedef long long int LL;
const int N = 100010;
vector<LL> triangular;
LL n, k;

void init() {
	triangular = vector<LL>(N, 0);
	for (int i = 1; i < N; i++)
		triangular[i] = triangular[i - 1] + i;
}

string solve() {
	int pos, b1, b2;
	string ans(n, 'a');
	pos = lower_bound(triangular.begin(), triangular.end(), k) - triangular.begin();
	b1 = n - pos - 1;
	b2 = b1 + 1 + (triangular[pos] - k);
	ans[b1] = ans[b2] = 'b';
	return ans;
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(nullptr);
	cout.tie(nullptr);
	int t;
	init();
	for (cin >> t; t; t--) {
		cin >> n >> k;
		cout << solve() << endl;
	}
}

File: since2020/CodeForces/1328B.py
'''
	@judge CodeForces
	@id 1328B
	@name K-th Beautiful String
	@contest Codeforces Round #629 (Div. 3)

	@tag Combination & Permutaion
'''
from sys import stdin
from bisect import bisect_left

N = 100010
triangular = [ 0 ]
for n in range(1, N):
	triangular.append(triangular[-1] + n)

def solve(n, k):
	pos = bisect_left(triangular, k)
	b1 = n - pos - 1
	b2 = b1 + triangular[pos] - k + 1
	return 'a' * b1 + 'b' + 'a' * (b2 - b1 - 1) + 'b' + 'a' * (n - b2 - 1)

input()
for line in stdin:
	n, k = map(int, line.split())
	print(solve(n, k))